### PR TITLE
implement skinned 1st person view hand

### PR DESCRIPTION
**Pull request by bell07**
I implement the skinning of 1st person view hand today, based on implementation https://github.com/jordan4ibanez/newhand/. The hand model is copied from there.

Since I am not model developer, maybe some optimizations are needed. The model does not support 1.8er skins, so it works for 1.0 only. 

Works great with fensta skins!

### CODE REVIEW

File: skin_meta_api.lua [hunk 1]
```diff
@@ -174,6 +201,14 @@ function skin_class:apply_skin_to_player(player)
 			y = self:get_meta("visual_size_y") or 1
 		}
 	})
+
+	local hand = self:get_hand()
+	if hand then
+		player:get_inventory():set_size("hand", 1)
+		player:get_inventory():set_stack("hand", 1, hand)
+	else
+		player:get_inventory():set_stack("hand", 1, "")
```
Comment (SmallJoker) [inline skin_meta_api.lua:210]: To ensure sanity, this list needs to be "write"-protected using `minetest.register_allow_player_inventory_action`. Even if this list is not exposed by a GUI, CSM mods can allow access to it.

File: skin_meta_api.lua [hunk 2]
```diff
@@ -51,6 +51,33 @@ function skin_class:get_texture()
...
+
+	minetest.register_node(hand, {
+		tiles = {self:get_texture()},
+		inventory_image = "wieldhand.png",
+		visual_scale = 1,
+		wield_scale = {x=1,y=1,z=1},
+		paramtype = "light",
+		drawtype = "mesh",
+		mesh = "skinsdb_hand.b3d",
+		use_texture_alpha = "clip",
```
[the remark this answers is not in the document]
Comment (bell07) [reply skin_meta_api.lua:69]: How the fallback usually looks like?

File: skin_meta_api.lua (hunk 1 above)
Comment (bell07) [reply skin_meta_api.lua:210]: Do I understand the minetest.register_allow_player_inventory_action() right  I just need to return 0? How I can test it?

File: skin_meta_api.lua (hunk 2 above)
[the remark this answers is not in the document]
Comment (SmallJoker) [reply skin_meta_api.lua:69]: https://github.com/minetest-mods/xdecor/commit/3b5b3f0c1

File: skin_meta_api.lua (hunk 1 above)
Comment (SmallJoker) [reply skin_meta_api.lua:210]: > How I can test it?

Add the slot to any formspec `list[current_player;hand;XX,YY;1,1;]`. It does not matter which one.

```lua
minetest.register_allow_player_inventory_action(function(player, action, inv, data)
	if data.to_list == "hand" or data.from_list == "hand" or data.listname == "hand" then
		return 1
	end
end)
```

File: API.md [hunk 3]
```diff
@@ -55,6 +55,15 @@ Get the skin texture for any reason. Note to apply them the skin:set_skin() shou
 
 Could be redefined for dynamic texture generation
 
+## skin:set_hand(hand_node)
+Set the hand node to be used with this skin
+
+## skin:set_hand_from_texture()
+Register and set hand node based on skin texture
```
[the remark this answers is not in the document]
Comment (bell07) [reply API.md:62]: resolved, but github does not mark the change sas outdated :-(

File: skin_meta_api.lua (hunk 1 above)
Comment (bell07) [reply skin_meta_api.lua:211]: Added the function into init.lua. Tested shortly without and with the code. Work as expected.

**Review by SmallJoker — APPROVED** [review]
LGTM

**Pull request merged**